workspaces/spectacle/src/spectacle.ts

```diff
@@ -197,7 +197,9 @@ export async function makeSpectacle(opticContext: IOpticContext) {
         args: any,
         context: GraphQLContext
       ) => {
-        return context.spectacleContext().shapeViewerProjection[args.shapeId];
+        return (
+          context.spectacleContext().shapeViewerProjection[args.shapeId] || []
```

Comment:
This was _sometimes_ undefined if we query with a shapeId not defined in the projection.